packages/backend/src/api/controllers/userTransactionToEntry.ts

```diff
@@ -0,0 +1,84 @@
+import { TransactionEntry } from '@explorer/frontend'
+import { Asset } from '@explorer/frontend/src/utils/assets'
+
+import { CollateralAsset } from '../../config/starkex/StarkexConfig'
+import { UserTransactionRecord } from '../../peripherals/database/transactions/UserTransactionRepository'
+import { assertUnreachable } from '../../utils/assertUnreachable'
+
+function extractUserTxAmount(
+  data: UserTransactionRecord['data']
+): bigint | undefined {
+  switch (data.type) {
+    case 'ForcedWithdrawal':
+      return data.quantizedAmount
+    case 'ForcedTrade':
+      return data.syntheticAmount
+    case 'FullWithdrawal':
+      return undefined
+    case 'Withdraw':
+      return data.quantizedAmount
+    case 'WithdrawWithTokenId':
+      return data.quantizedAmount
+    case 'MintWithdraw':
+      return data.quantizedAmount
+    default:
+      assertUnreachable(data)
+  }
+}
+
+function extractUserTxAsset(
+  data: UserTransactionRecord['data'],
+  collateralAsset?: CollateralAsset
+): Asset | undefined {
+  switch (data.type) {
+    case 'ForcedWithdrawal':
+      return collateralAsset ? { hashOrId: collateralAsset.assetId } : undefined
+    case 'ForcedTrade':
+      return { hashOrId: data.syntheticAssetId }
+    case 'FullWithdrawal':
+      return collateralAsset ? { hashOrId: collateralAsset.assetId } : undefined
+    case 'Withdraw':
+      return { hashOrId: data.assetType }
+    case 'WithdrawWithTokenId':
+      return { hashOrId: data.assetId }
+    case 'MintWithdraw':
+      return { hashOrId: data.assetId }
+    default:
+      assertUnreachable(data)
+  }
+}
+
+function extractUserTxEntryType(
+  data: UserTransactionRecord['data']
+): TransactionEntry['type'] {
+  switch (data.type) {
+    case 'ForcedWithdrawal':
+      return 'FORCED_WITHDRAW'
+    case 'ForcedTrade':
+      return data.isABuyingSynthetic ? 'FORCED_BUY' : 'FORCED_SELL'
+    case 'FullWithdrawal':
+      return 'FORCED_WITHDRAW'
+    case 'Withdraw':
+      return 'WITHDRAW'
+    case 'WithdrawWithTokenId':
+      return 'WITHDRAW'
+    case 'MintWithdraw':
+      return 'WITHDRAW'
```

Comment:
We don't have to return `WITHDRAW` 3 times. We can have just one return here

Reply:
Fixed